logging/logging.go

```diff
@@ -0,0 +1,69 @@
+package logging
+
+import (
+	"os"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+)
+
+// Interface to mock os.Getenv() operations
+type envOperations interface {
+	Getenv(key string) string
+}
+
+type realEnvOperations struct{}
+
+func (r *realEnvOperations) Getenv(key string) string {
+	return os.Getenv(key)
+}
+
+// Unfortunately LogLevel() in terraform plugin sdk v2 is not test friendly so we have to mock it since it calls os.GetEnv() underneath
+type logLevelGetter interface {
+	GetLogLevel() string
+}
+
+type realLogLevelGetter struct{}
+
+func (r *realLogLevelGetter) GetLogLevel() string {
+	return logging.LogLevel()
+}
+
+func SetLogger() {
+	logger.DefaultLogger = &logger.SimpleLogger{
+		Level: terraformToSDKLogLevel(&realEnvOperations{}, &realLogLevelGetter{}),
+	}
+}
+
+// Golang doesn't have mutable maps, having global mutable variable is less safe as compared to a function returning map
+func terraformToSDKLogLevelMapping() map[string]logger.Level {
+	return map[string]logger.Level{
+		"TRACE": logger.LevelTrace,
+		"DEBUG": logger.LevelDebug,
+		"INFO":  logger.LevelInfo,
+		"WARN":  logger.LevelWarn,
+		"ERROR": logger.LevelError,
+	}
+}
+
+func terraformToSDKLogLevel(envOps envOperations, logGetter logLevelGetter) logger.Level {
+	// Exception for unknown log levels are handled in underlying LogLevel() method
```

Comment:
Interesting. This probably works fine given the current SDK logging architecture and gets us back to where we were, but I think it's a bit of a hack to try to go around TF logging infra and read from the environment like this. For instance, users can't enable debug logging for just our provider (otherwise, we would also need to check TF_LOG_PROVIDER_DATABRICKS as well).

Another approach would be to define a struct and implement the logging.Logger interface, then configure the SDK to use this logger. This is what is done in the AWS TF provider implementation. They [implement a custom logger](https://github.com/hashicorp/aws-sdk-go-base/blob/main/logging/tf_logger.go) that implements their [Logger interface](https://github.com/hashicorp/aws-sdk-go-base/blob/main/logging/logger.go) and then [configure the client to use that logger at configuration time](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/conns/config.go#L64). The advantage of this approach is that you let TF determine what gets printed to the log (so, for instance, TF could change the log level while the provider was running, or provide different ways for users to specify log level rather than through environment variables).